xbmc/video/ContextMenus.cpp

```diff
@@ -58,6 +58,8 @@ bool CMarkWatched::IsVisible(const CFileItem& item) const
   {
     if (item.HasVideoInfoTag())
       return item.IsVideoDb();
+    else if (item.GetProperty("IsVideoFolder").asBoolean())
```

Comment:
Not sure I understand what you mean. The problem is that item can be anything (!) and we must only return true in case of certain items - namely videos/recordings and video folders/recording folders.